### ISSUE TITLE
Popover will not close if it starts out as open

### ISSUE THREAD

**Issue by dcwither** [littlebits/react-popover#83]
If the popover is initialized with `isOpen={true}` then later setting `isOpen={false}` does not close the popover.

After some investigation this occurs because `state.toggle` starts out as false, so switching `isOpen` to false does not change toggle and it doesn't trigger any animation.

Running the following example and clicking close will result in the popover remaining open.

``` javascript
import Popover from 'react-popover';

React.createClass({
  getInitialState() {
    return {
      isOpen: true
    };
  },

  handleClose() {
    this.setState({isOpen: false});
  },

  render() {
    return <Popover isOpen={this.state.isOpen}>
      <button onClick={this.handleClose}>Close</button>
    </Popover>
  }
});
```

**Comment by brownsmith** [comment]
Can you set the state in componentWillMount?

**Comment by dcwither** [comment]
yes, there's definitely work arounds. It just seems to me that this results in an inconsistent internal state, and should be flagged.

**Comment by oyeanuj** [comment]
@jasonkuhrt I am also facing this problem and in my case, the workaround above doesn't work either. In my case, the `isOpen` is being set by parent props (so, hence during render time and not in the constructor). And it isn't easy or clean to construct this and then pass in the props later.

FWIW, here is a screenshot of the props and state of the popover at the time when it should have closed. 
<img width="295" alt="screen shot 2017-03-05 at 12 10 12 pm" src="https://cloud.githubusercontent.com/assets/9633371/23590970/f965f082-019d-11e7-9609-4a7d1e648831.png">

**Comment by jasonkuhrt** [comment]
Fixed in #97. Released in https://github.com/littlebits/react-popover/releases/tag/0.4.8

Thanks @pasviegas for the PR!
